pkg/loadbalancer/redirectpolicy/controller.go

```diff
@@ -379,9 +379,14 @@ func (c *lrpController) updateRedirectBackends(wtxn writer.WriteTxn, ws *statedb
 				continue
 			}
 			beps = append(beps, lb.BackendParams{
-				Address:   addr.L3n4Addr,
-				State:     lb.BackendStateActive,
-				PortNames: []string{addr.portName},
+				Address: addr.L3n4Addr,
+				State:   lb.BackendStateActive,
+				PortNames: func() []string {
+					if addr.portName != "" {
+						return []string{addr.portName}
+					}
+					return []string{}
```

Comment:
could return `nil` to avoid allocating.

Reply:
That’s what was returned in the earlier version, but I decided to pass an empty slice instead to avoid causing a panic in callers. I don't reverting to the earlier version if we are confident that it won't cause a panic in production.